src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -50,7 +48,7 @@ internal class MissingSetupComponentRegistrationService : IMissingSetupComponent
         private IVsSolution? _vsSolution;
         private bool? _isVSFromPreviewChannel;
 
-        private readonly object _lock = new();
+        private static readonly object _lock = new();
```

Comment:
I don't think this should be static as you don't want to share a lock between instances of the same class. Each should have their own lock.

Reply:
removing statis since this is a service.